### PR TITLE
Introduce ability to version Media

### LINKED ISSUE #24410: Ability to add versions to js and css files
Hello guys,
#### Describe the bug
when we change css and js files for a module, we need to add a version to update clients caches to get up to dated files.
There is a need to specify the version when we register js and css files with these methods in controllers: 

- **registerJavascript**
- **registerStylesheet**

#### Steps to reproduce

1. Fresh install
2. Make sure the CCC is disabled
3. Enable Debug mode
4. Change anything in CSS/JS
5. Changes aren't visible on front for client

#### Additional information

* PrestaShop version: 1.7.7.x
* PHP version: 7.3

### PR DESCRIPTION
**Pull request by samberrry**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | This PR adds the ability to add versions to js and css files in modules. In some cases module changes will not be applied due to client's cache and needs to add a version to js/css files. 
| Type?             | new feature
| Category?         | FO 
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | #24410
| How to test?      | Add "version" property to params in registerStylesheet and registerJavascript methods 
| Possible impacts? | nothing

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24409)

### CODE REVIEW

**Comment by samberrry** [comment]
@Progi1984 @kpodemski new PR.

File: classes/assets/JavascriptManager.php [hunk 1]
```diff
@@ -55,6 +55,7 @@ protected function getDefaultList()
      * @param bool $inline
      * @param string $attribute
      * @param string $server
+     * @param string $version
```
Comment (Progi1984) [inline classes/assets/JavascriptManager.php]:
```suggestion
     * @param string|null $version
```

File: classes/assets/JavascriptManager.php [hunk 2]
```diff
@@ -93,13 +95,16 @@ public function unregisterById($idToRemove)
...
```
Comment (Progi1984) [inline classes/assets/JavascriptManager.php]:
```suggestion
     * @param string|null $version
```

File: classes/assets/StylesheetManager.php [hunk 3]
```diff
@@ -54,6 +54,7 @@ protected function getDefaultList()
      * @param bool $inline
      * @param string $server
      * @param bool $needRtl
+     * @param string $version
```
Comment (Progi1984) [inline classes/assets/StylesheetManager.php]:
```suggestion
     * @param string|null $version
```

File: classes/assets/StylesheetManager.php [hunk 4]
```diff
@@ -106,12 +108,15 @@ public function getList()
      * @param int $priority
      * @param bool $inline
      * @param string $server
+     * @param string $version
```
Comment (Progi1984) [inline classes/assets/StylesheetManager.php]:
```suggestion
     * @param string|null $version
```

File: classes/assets/StylesheetManager.php [hunk 5]
```diff
@@ -62,18 +63,19 @@ public function register(
         $priority = self::DEFAULT_PRIORITY,
         $inline = false,
         $server = 'local',
-        $needRtl = true
+        $needRtl = true,
+        string $version = null
```
Comment (Progi1984) [inline classes/assets/StylesheetManager.php]:
```suggestion
        ?string $version = null
```

**Review by Progi1984 — COMMENTED** [review]
Last comments, and that should be ok

**Review by Progi1984 — DISMISSED** [review]

**Comment by Progi1984** [comment]
@samberrry Could you fix Github Actions, please ?

**Review by PierreRambaud — DISMISSED** [review]

**Comment by kpodemski** [comment]
@samberrry you can run `php vendor/bin/php-cs-fixer fix /classes/assets` and it should be enough 👍

**Review by kpodemski — CHANGES_REQUESTED** [review]
code style to fix

File: classes/controller/FrontController.php [hunk 6]
```diff
@@ -1077,6 +1077,8 @@ public function registerStylesheet($id, $relativePath, $params = [])
             'priority' => AbstractAssetManager::DEFAULT_PRIORITY,
             'inline' => false,
             'server' => 'local',
+            'version' => null,
+            'needRtl' => true
```
Comment (PululuK) [inline classes/controller/FrontController.php]:
```suggestion
            'needRtl' => true,
```

Comment (samberrry) [reply classes/controller/FrontController.php]: @PululuK thank you

**Review by PululuK — DISMISSED** [review]

**Review by kpodemski — DISMISSED** [review]
Well done. LGTM

File: classes/controller/FrontController.php (hunk 6 above)
Comment (PululuK) [reply classes/controller/FrontController.php]: I use my custom php-cs-fixer script https://gist.github.com/PululuK/07ac93607dd6a30c90f9034f32117dcd

**Review by PierreRambaud — DISMISSED** [review]

**Comment by Robin-Fischer-PS** [comment]
Can this be tested by a maintainer ? Thanks :D

**Review by jolelievre — APPROVED** [review]
I just rebased the PR but didn't change the code
I tested the PR here is the QA sample https://drive.google.com/file/d/1wu6dFeVT2aCBUi7Yb8Q6h4T16A64iOoU/view

- URLs are correctly updated based on the version parameter
- the cache is correctly used, and updated when needed

Works like a charm!! Good job!
Thank you @samberrry

**Review by kpodemski — APPROVED** [review]

**Comment by samberrry** [comment]
@kpodemski @jolelievre thank you all

**Pull request merged**